packages/grid/_modules_/grid/hooks/features/pagination/useGridPageSize.ts

```diff
@@ -20,14 +20,22 @@ import { gridDensityRowHeightSelector } from '../density';
  */
 export const useGridPageSize = (
   apiRef: GridApiRef,
-  props: Pick<GridComponentProps, 'pageSize' | 'onPageSizeChange' | 'autoPageSize'>,
+  props: Pick<
+    GridComponentProps,
+    'pageSize' | 'onPageSizeChange' | 'autoPageSize' | 'initialState'
+  >,
 ) => {
   const logger = useGridLogger(apiRef, 'useGridPageSize');
   const rowHeight = useGridSelector(apiRef, gridDensityRowHeightSelector);
 
   useGridStateInit(apiRef, (state) => ({
     ...state,
-    pagination: { pageSize: props.pageSize ?? (props.autoPageSize ? 0 : 100) },
+    pagination: {
+      pageSize:
```

Comment:
Can we move complex ternary operators like this one outside into a variable? It's easier to read.

Reply:
Done :+1: